hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ReconContainerMetadataManagerImpl.java

```diff
@@ -515,4 +547,84 @@ public void commitBatchOperation(RDBBatchOperation rdbBatchOperation)
       throws IOException {
     this.containerDbStore.commitBatchOperation(rdbBatchOperation);
   }
+    
+  /**
+   * Use the DB's prefix seek iterator to start the scan from the given
+   * key prefix and key version.
+   *
+   * @param keyPrefix the given keyPrefix.
+   * @param keyVersion the given keyVersion.
+   * @return Map of (KeyPrefixContainer, Integer).
+   */
+  @Override
+  public Map<KeyPrefixContainer, Integer> getContainerForKeyPrefixes(
+      String keyPrefix, long keyVersion) throws IOException {
+
+    Map<KeyPrefixContainer, Integer> containers = new LinkedHashMap<>();
+    TableIterator<KeyPrefixContainer, ? extends KeyValue<KeyPrefixContainer,
+        Integer>> keyIterator = keyContainerTable.iterator();
```

Comment:
Need to close DB table iterator after use.

Either do `keyIterator.close()` before each `return`, or use try-with-resource (recommended).

Reply:
Updated with try-with-resource.